### PR TITLE
[ACCEL] Change staff access role

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
Change staff access role.
"staff" role has instructor permissions in LMS and doesn't have access to course in CMS.

**Youtrack:**
https://youtrack.raccoongang.com/issue/ACCEL-183

### CODE REVIEW

File: common/djangoapps/student/auth.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def get_user_permissions(user, course_key, org=None):
         return all_perms
     # Staff have all permissions except EDIT_ROLES:
     if OrgStaffRole(org=org).has_user(user) or (course_key and user_has_role(user, CourseStaffRole(course_key))):
-        return STUDIO_VIEW_USERS | STUDIO_EDIT_CONTENT | STUDIO_VIEW_CONTENT
+        return STUDIO_VIEW_USERS
```
Comment (idegtiarov) [inline common/djangoapps/student/auth.py]: Please add a comment to this change

File: lms/djangoapps/courseware/views/views.py [hunk 2]
```diff
@@ -980,6 +981,15 @@ def _progress(request, course_key, student_id):
     # checking certificate generation configuration
     enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(student, course_key)
 
+    user_role = CourseAccessRole.objects.filter(
+        user=request.user,
+        course_id=course_key
+    ).values_list('role', flat=True)
+
+    is_local_staff = True
```
Comment (idegtiarov) [inline lms/djangoapps/courseware/views/views.py]: Please describe in the comment what does local_staff mean?

File: openedx/core/lib/xblock_utils/__init__.py [hunk 3]
```diff
@@ -318,12 +319,21 @@ def add_staff_markup(user, disable_staff_debug_info, block, view, frag, context)
             # build edit link to unit in CMS. Can't use reverse here as lms doesn't load cms's urls.py
             edit_link = "//" + settings.CMS_BASE + '/container/' + text_type(block.location)
 
+            user_role = CourseAccessRole.objects.filter(
+                user=user,
+                course_id=block.course_id
+            ).values_list('role', flat=True)
+
+            is_local_staff = True
```
Comment (idegtiarov) [inline openedx/core/lib/xblock_utils/__init__.py]: ditto

File: lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html [hunk 4]
```diff
@@ -104,7 +104,7 @@
         <section class="instructor-dashboard-content-2" id="instructor-dashboard-content">
           <h2 class="hd hd-2 instructor-dashboard-title">${_("Instructor Dashboard")}</h2>
           <div class="wrap-instructor-info studio-view">
-            %if studio_url:
+            %if studio_url and not sections[0].get('access').get('staff') or request.user.is_staff:
```
Comment (idegtiarov) [inline lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html]: Could you please clarify what is expected in the statement?

Comment (arsentur) [reply lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html]: The link will be shown only if the user has is_staff or has 'admin' access and a studio URL exists. And the link will not be displayed if the user has the role “staff” or if he is a student.

File: common/djangoapps/student/auth.py (hunk 1 above)
Comment (arsentur) [reply common/djangoapps/student/auth.py]: Done

File: lms/djangoapps/courseware/views/views.py (hunk 2 above)
Comment (arsentur) [reply lms/djangoapps/courseware/views/views.py]: Done

File: openedx/core/lib/xblock_utils/__init__.py (hunk 3 above)
Comment (arsentur) [reply openedx/core/lib/xblock_utils/__init__.py]: Done

File: lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html (hunk 4 above)
Comment (idegtiarov) [reply lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html]: Now statement works in a little bit different way, without studio_url, and with request.user.is_staff the statement returns True which looks a little bit odd.

Comment (arsentur) [reply lms/templates/instructor/instructor_dashboard_2/instructor_dashboard_2.html]: Done

**Pull request merged**